app/assets/javascripts/test/sagas/saga_integration.spec.js

```diff
@@ -0,0 +1,42 @@
+import test from "ava";
+import mockRequire from "mock-require";
+import { setupOxalis, TIMESTAMP } from "test/helpers/apiHelpers";
+import update from "immutability-helper";
+import Store from "oxalis/store";
+import { wkReadyAction } from "oxalis/model/actions/actions";
+import { createSaveQueueFromUpdateActions } from "test/helpers/saveHelpers";
+import { generateTreeName } from "oxalis/model/reducers/skeletontracing_reducer_helpers";
+import { getStats } from "oxalis/model/accessors/skeletontracing_accessor";
+import Utils from "libs/utils";
+
+const UpdateActions = mockRequire.reRequire("oxalis/model/sagas/update_actions");
+
+test.beforeEach(async t => {
+  // Setup oxalis, this will execute model.fetch(...) and initialize the store with the tracing, etc.
+  await setupOxalis(t, "task");
+  // Dispatch the wkReadyAction, so the sagas are started
+  Store.dispatch(wkReadyAction());
+});
+
+test("watchTreeNames saga should rename empty trees in tasks and these updates should be persisted", t => {
+  const state = Store.getState();
+  const treeWithEmptyName = state.tracing.trees[1];
+  const treeWithCorrectName = update(treeWithEmptyName, {
+    name: { $set: generateTreeName(state, treeWithEmptyName.timestamp, treeWithEmptyName.treeId) },
+  });
+
+  const saveQueue = createSaveQueueFromUpdateActions(
```

Comment:
Can you rename this to `expectedSaveQueue`? I was a bit confused at first why this is manually defined :)